lib/trie/version.go

```diff
@@ -55,7 +55,17 @@ func (v Version) ShouldHashValue(value []byte) bool {
 var ErrParseVersion = errors.New("parsing version failed")
 
 // ParseVersion parses a state trie version string.
-func ParseVersion(s string) (version Version, err error) {
+func ParseVersion[T string | uint32](v T) (version Version, err error) {
+	var s string
+	switch value := any(v).(type) {
+	case string:
+		s = value
+	case uint32:
+		s = fmt.Sprintf("V%d", value)
+	default:
+		panic(fmt.Sprintf("unsupported type %T", s))
```

Comment:
This default case is useless since the type definitions in generic only defines `string | uint32` any other type will be denied at comptime

```suggestion
```

Reply:
True! thanks!!